module/VuFind/src/VuFind/Auth/ILSAuthenticator.php

```diff
@@ -336,8 +338,19 @@ public function storedCatalogLogin()
                 $this->ilsAccount[$username] = $patron;
                 return $patron;
             }
+        } elseif (!empty($user_name)) {
+            $user = $this->getDbService(UserServiceInterface::class)->getUserByUsername($user_name);
```

Comment:
Should you do this lookup after checking the cache, so you don't make an unnecessary database call?

Reply:
Also, should you account for the possibility that no matching user is found?

Reply:
should be done